src/main/java/slimeknights/tconstruct/smeltery/tileentity/AbstractCastingTileEntity.java

```diff
@@ -60,7 +60,7 @@ public AbstractCastingTileEntity(TileEntityType<?> tileEntityTypeIn, IRecipeType
     this.tank = new FluidTankAnimated(0, this);
     this.itemHandler = new SidedInvWrapper(this, Direction.DOWN);
     this.recipeType = recipeType;
-    this.crafting = new TileCastingWrapper(this, Fluids.EMPTY);
+    this.crafting = new TileCastingWrapper(this, Fluids.EMPTY, null);
```

Comment:
Might be better to make the constructor just `this`, set the other two by default.